design/authn_azure/authn_azure_solution_design.md

```diff
@@ -0,0 +1,357 @@
+the Azure VM resource will receive a Conjur access token in return. All further requests that this Azure VM sends will need to have this Conjur access token in the header.
+
+##### Host annotation / xms_mirid mapping
+
+Recap, Conjur Host annotations that need to be filled and match based on "xms_mirid" claim in the Azure token:
+
+```yaml
+  authn-azure/subscription-id: test-subscription 
+  authn-azure/resource-group: test-group 
+  authn-azure/user-assigned-identity: test-app-pipeline 
+OR 
+  authn-azure/system-assigned-identity: ie_qWCXhXxt1zIEsu4c7acQVGn4
+```
+
+An example of a "xms_mirid" claim is listed below. We will use what is defined in this claim and compare it against the Conjur identity for the Azure resource. If they match, then we can confirm that the 
+Azure resource that has made the request has a Conjur identity.
+
+```yaml
+"xms_mirid": "/subscriptions/<subscription_id>/resourcegroups/<resource_group>/providers/Microsoft.ContainerInstance/containerGroups/<assigned_identity>"
+```
+
+| Host annotation                                                      | xms_mirid              |
+|----------------------------------------------------------------------|------------------------|
+| `authn-azure/subscription-id`                                        | subscription_id        |
+| `authn-azure/resource-group`                                         | resource_group         |
+| `authn-azure/user-assigned-identity` or `system-assigned-identity`   | assigned_identity      |
+
+### Backwards compatibility
+
+None because this is a new feature.
+
+### Performance
+
+Azure authenticator performance should conform with our other authenticators with an average call time of 1 second.
+
+### Affected Components
+
+- Conjur
+
+- DAP
+
+## Test Plan
+
+TBT
+
+### Integration tests
+
+#### Functionality
+
+This suite of tests assumes that the Azure authenticator has been configured correctly
+
+|    | Given                                                                                        | When                                                     | Then                                                                                          | Manual / UT / Integration | Status     |
+|----|----------------------------------------------------------------------------------------------|----------------------------------------------------------|-----------------------------------------------------------------------------------------------|---------------------------|------------|
+| 1  | *Vanilla flow* - a Conjur identity exists for Azure VM and added to privileged layer         | VM sends authentication request with a valid Azure token | Conjur access token is returned in body and action is logged                                  | Integration               |            |
```

Comment:
VM sends authentication request with an **invalid** Azure token

Reply:
I check that in number 7 ( One or more fields in Azure token is missing which is an invalid token